### PR TITLE
unique usernames

### PR DESCRIPTION
**Pull request by aweolumidedavid**
- [x] issue #105
- [x] follows contribution [guide](https://github.com/keep-starknet-strange/art-peace/blob/main/CONTRIBUTING.md)
- [x] code change includes tests
- [x] breaking change

### CODE REVIEW

**Comment by mubarak23** [comment]
@aweolumidedavid fo the` build/ check` to pass, 
cd into onchain directory and 
run `scarb fmt`, and push the changes,

File: onchain/src/tests/username_store.cairo [hunk 1]
```diff
@@ -14,23 +16,39 @@ fn deploy_contract() -> ContractAddress {
...
 
-    let username_address = dispatcher.get_username('devsweet');
+    let new_username_address = dispatcher.get_username_address(new_username);
+    assert!(new_username_address == utils::PLAYER1(), "New username not linked");
+    let new_username = dispatcher.get_username(utils::PLAYER1());
+    assert!(new_username == new_username, "New username not set");
 
-    assert(username_address == second_contract_address, 'Username not Transferred');
+    let old_username_address = dispatcher.get_username_address(initial_username);
+    assert!(old_username_address == contract_address_const::<0>(), "Old username not unlinked");
```
Comment (b-j-roberts) [inline onchain/src/tests/username_store.cairo:53]: Can you add a couple tests that check that a user cannot claim 2 usernames, and check that the user cannot change usernames when they dont have one ( use the `should_panic` option )

**Review by b-j-roberts — CHANGES_REQUESTED** [review]
Thank you, just a couple more tests I'd like

File: onchain/src/tests/username_store.cairo (hunk 1 above)
Comment (aweolumidedavid) [reply onchain/src/tests/username_store.cairo:53]: No problem, I am on it

Comment (b-j-roberts) [reply onchain/src/tests/username_store.cairo:63]: Sorry, I meant to add 2 new tests not change the existing test.
So there should be 4 tests : `test_claim_username`, `test_change_username` and the new tests `test_cannot_claim_multiple_usernames` and `test_cannot_change_with_no_username`. That way we are testing all the different possible usages and possible problems.

**Comment by b-j-roberts** [comment]
Thank you for making those changes!

**Pull request merged**